### PR TITLE
Add RAG Ollama application guide to the documentation

### PR DESCRIPTION
**Pull request by mfranzon**
## Reviews

- [ ] Technical review
- [ ] Editorial review
- [ ] Product review

### CODE REVIEW

File: content/guides/use-case/rag-ollama/containerize.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+In the terminal, press `ctrl`+`c` to stop the application.
+
+## Summary
+
+In this section, you learned how you can containerize and run your RAG
+application using Docker.
+
+## Next steps
+
+In the next section, you'll learn how you can run your application, database, and LLM service all locally using Docker.
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:82]: At this point, it looks like the db and ollama are already running locally. Suggest rewording this for this specific app.

**Review by craig-osterhout — CHANGES_REQUESTED** [review]
Thanks @mfranzon, it's looking good.

You can update the `data/toc.yaml` so the topics appear in the table of contents. Also, you can update the use-case guides landing page at `content/guides/use-case/_index.md` to include this guide. 

In general, it looks like your sample repo already has everything, so I suggest to avoid telling the user to create/add things because they might overwrite files or add duplicate instructions without realizing it's all already there.

File: content/guides/use-case/rag-ollama/containerize.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+server-1  |   You can now view your Streamlit app in your browser.
+server-1  |
+server-1  |   URL: http://0.0.0.0:8501
+server-1  |
+```
+
+Open a browser and view the application at [http://localhost:8501](http://localhost:8501). You should see a simple Streamlit application. 
+
+The application requires a Qdrant database service and an LLM service to
+function. If you have access to services that you ran outside of Docker, specify
```
[the remark this answers is not in the document]
Comment (mfranzon) [reply content/guides/use-case/rag-ollama/containerize.md]: add more info about env variables

File: content/guides/use-case/rag-ollama/develop.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+- Adding a local database
+- Adding a local or remote LLM service
+
+> **Note**
+>
+> You can see more samples of containerized GenAI applications in the [GenAI Stack](https://github.com/docker/genai-stack) demo applications.
+
+## Add a local database
+
+You can use containers to set up local services, like a database. In this section, you'll update the `compose.yaml` file to define a database service. In addition, you'll specify an environment variables file to load the database connection information rather than manually entering the information every time.
```
[the remark this answers is not in the document]
Comment (mfranzon) [reply content/guides/use-case/rag-ollama/develop.md]: accept the rephrasing

File: content/guides/use-case/rag-ollama/containerize.md (hunk 1 above)
Comment (mfranzon) [reply content/guides/use-case/rag-ollama/containerize.md]: rephrase this section

**Comment by mfranzon** [comment]
All the required changes are in place. Let me know if I had to change something else, thank you!

File: content/guides/use-case/_index.md [hunk 4]
```diff
@@ -37,6 +37,10 @@ grid_genai:
   description: Explore an app that can summarize text.
   link: /guides/use-case/nlp/text-summarization/
   icon: summarize
+- title: RAG Ollama application
+  description: Explore how to containerize a RAG application
```
Comment (craig-osterhout) [inline content/guides/use-case/_index.md:41]:
```suggestion
  description: Explore how to containerize a RAG application.
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+│ └── README.md
+```
+
+## Containerizing your application: Essentials
+
+Containerizing an application involves packaging it along with its dependencies into a container, which ensures consistency across different environments. Here’s what you need to containerize an app like Winy :
+
+1. Dockerfile: A Dockerfile that contains instructions on how to build a Docker image for your application. It specifies the base image, dependencies, configuration files, and the command to run your application.
+
+2. Docker Compose File: Docker Compose is a tool for defining and running multi-container Docker applications. A Compose file, typically named docker-compose.yml, allows you to configure your application's services, networks, and volumes in a single file.
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:53]: Suggest removing the part about the typical name as it is no longer the preferred name.

```suggestion
2. Docker Compose File: Docker Compose is a tool for defining and running multi-container Docker applications. A Compose file allows you to configure your application's services, networks, and volumes in a single file.
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+container_name: winy 
+ports:
+    - "8501:8501" 
+depends_on:
+    - qdrant 
+    - ollama 
+```
+
+If you don't have the services running, continue with this guide to learn how you can run some or all of these services with Docker.
+Remember that the `ollama` service is empty, it has not any model, for this reason you had to pull a model before starting to use the RAG application. All the instruction are in the following page!
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:94]:
```suggestion
Remember that the `ollama` service is empty; it doesn't have any model. For this reason you need to pull a model before starting to use the RAG application. All the instructions are in the following page.
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+The sample application supports both [Ollama](https://ollama.ai/). This guide provides instructions for the following scenarios:
+- Run Ollama in a container
+- Run Ollama outside of a container
+
+While all platforms can use any of the previous scenarios, the performance and
+GPU support may vary. You can use the following guidelines to help you choose the appropriate option:
+- Run Ollama in a container if you're on Linux, and using a native installation of the Docker Engine, or Windows 10/11, and using Docker Desktop, you
+  have a CUDA-supported GPU, and your system has at least 8 GB of RAM.
+- Run Ollama outside of a container on a Linux Machine.
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:69]:
```suggestion
- Run Ollama outside of a container if running Docker Desktop on a Linux Machine.
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+                count: 1
+                capabilities: [gpu]
+   ```
+
+   > **Note**
+   >
+   > For more details about the Compose instructions, see [Turn on GPU access with Docker Compose](../../../compose/gpu-support.md).
+
+3. Once the Ollama container is up and running it is possible to use the `download_model.sh` inside the `tools` folder with this command:
+```console
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:105]:
```suggestion
   ```console
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+. ./download_model.sh <model-name>
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:106]:
```suggestion
   . ./download_model.sh <model-name>
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+```
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:107]:
```suggestion
   ```
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+To run Ollama outside of a container:
+1. [Install](https://github.com/jmorganca/ollama) and run Ollama on your host
+   machine.
+2. Pull the model to Ollama using the following command.
+   ```console
+   $ ollama pull llama2
+   ```
+3. Remove the `ollama` service from the `docker-compose.yaml` and update properly the connection variables in `winy` service:
+
+```diff
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:122]:
```suggestion
   ```diff
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+- OLLAMA=http://ollama:11434
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:123]:
```suggestion
   - OLLAMA=http://ollama:11434 
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+   machine.
+2. Pull the model to Ollama using the following command.
+   ```console
+   $ ollama pull llama2
+   ```
+3. Remove the `ollama` service from the `docker-compose.yaml` and update properly the connection variables in `winy` service:
+
+```diff
+- OLLAMA=http://ollama:11434 
++ OLLAMA=<your-url>
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:124]:
```suggestion
   + OLLAMA=<your-url>
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+```
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:125]:
```suggestion
   ```
```

File: content/guides/use-case/rag-ollama/develop.md [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+{{< tab name="Run Ollama outside of a container" >}}
+
+To run Ollama outside of a container:
+1. [Install](https://github.com/jmorganca/ollama) and run Ollama on your host
+   machine.
+2. Pull the model to Ollama using the following command.
+   ```console
+   $ ollama pull llama2
+   ```
+3. Remove the `ollama` service from the `docker-compose.yaml` and update properly the connection variables in `winy` service:
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/develop.md:120]:
```suggestion
3. Remove the `ollama` service from the `docker-compose.yaml` and update the connection variables in `winy` service:
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+server-1  |
+```
+
+Open a browser and view the application at [http://localhost:8501](http://localhost:8501). You should see a simple Streamlit application. 
+
+The application requires a Qdrant database service and an LLM service to work properly. If you have access to services that you ran outside of Docker, specify the connection information in the `docker-compose.yaml`.
+
+```yaml
+winy:
+build:
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:79]:
```suggestion
  build:
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+    context: ./app 
+    dockerfile: Dockerfile
+environment:
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:82]:
```suggestion
  environment:
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+    - QDRANT_CLIENT=http://qdrant:6333 # Specifies the url for the qdrant database
+    - OLLAMA=http://ollama:11434 # Specifies the url for the ollama service
+container_name: winy
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:85]:
```suggestion
  container_name: winy 
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+```yaml
+winy:
+build:
+    context: ./app 
+    dockerfile: Dockerfile
+environment:
+    - QDRANT_CLIENT=http://qdrant:6333 # Specifies the url for the qdrant database
+    - OLLAMA=http://ollama:11434 # Specifies the url for the ollama service
+container_name: winy 
+ports:
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:86]:
```suggestion
  ports:
```

File: content/guides/use-case/rag-ollama/containerize.md [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+    - "8501:8501" 
+depends_on:
```
Comment (craig-osterhout) [inline content/guides/use-case/rag-ollama/containerize.md:88]:
```suggestion
  depends_on:
```

**Review by craig-osterhout — CHANGES_REQUESTED** [review]
Thanks, @mfranzon. Flow-wise, it looks good. Some of the spacing got misaligned and I added a few nit polish comments. Otherwise, LGTM.

**Comment by mfranzon** [comment]
Thank you very much @craig-osterhout for the review and the refinement, I have committed all your suggestion. If you think it's ready to merge, I don't have anything to add.

**Comment by dvdksn** [comment]
We recently did some restructuring in the docs repository. I've opened #20811 to carry this guide in, after rebasing and making the necessary updates as a result of the restructuring.
